Kernel/Syscalls/mmap.cpp

```diff
@@ -69,8 +69,12 @@ static bool should_make_executable_exception_for_dynamic_loader(bool make_readab
     return true;
 }
 
-static bool validate_mmap_prot(int prot, bool map_stack, bool map_anonymous, Memory::Region const* region = nullptr)
+static bool validate_mmap_prot(const Process& process, int prot, bool map_stack, bool map_anonymous, Memory::Region const* region = nullptr)
```

Comment:
This function can be moved into Process if we need access to it anyways.

Reply:
Moved both validation methods which use Process into Process (validate_mmap_prot and validate_inode_mmap_prot)